python/lsst/ip/diffim/subtractImages.py

```diff
@@ -0,0 +1,535 @@
+
+
+class AlardLuptonSubtractConfig(lsst.pipe.base.PipelineTaskConfig,
+                                pipelineConnections=AlardLuptonSubtractConnections):
+    mode = lsst.pex.config.ChoiceField(
+        dtype=str,
+        default="auto",
+        allowed={"auto": "Choose which image to convolve at runtime.",
+                 "convolveScience": "Only convolve the science image.",
+                 "convolveTemplate": "Only convolve the template image."},
+        doc="Choose which image to convolve at runtime, or require that a specific image is convolved."
+    )
+    makeKernel = lsst.pex.config.ConfigurableField(
+        target=lsst.ip.diffim.MakeKernelTask,
+        doc="Task to construct a matching kernel for convolution.",
+    )
+    doDecorrelation = lsst.pex.config.Field(
+        dtype=bool,
+        default=True,
+        doc="Perform diffim decorrelation to undo pixel correlation due to A&L "
+        "kernel convolution? If True, also update the diffim PSF."
+    )
+    decorrelate = lsst.pex.config.ConfigurableField(
+        target=lsst.ip.diffim.DecorrelateALKernelTask,
+        doc="Task to decorrelate the image difference.",
+    )
+    requiredTemplateFraction = lsst.pex.config.Field(
+        dtype=float,
+        default=0.1,
+        doc="Abort task if template covers less than this fraction of pixels."
+        "Setting to 0 will always attempt image subtraction."
+    )
+    doScaleVariance = lsst.pex.config.Field(
+        dtype=bool,
+        default=True,
+        doc="Scale variance of the image difference."
+    )
+    scaleVariance = lsst.pex.config.ConfigurableField(
+        target=ScaleVarianceTask,
+        doc="Subtask to rescale the variance of the template to the statistically expected level."
+    )
+    doApplyFinalizedPsf = lsst.pex.config.Field(
+        doc="Apply finalized psf models and aperture correction map.",
+        dtype=bool,
+        default=False,
+    )
+    doSubtractBackground = lsst.pex.config.Field(
+        doc="Subtract the background fit when solving the kernel?",
+        dtype=bool,
+        default=True,
+    )
+
+    forceCompatibility = lsst.pex.config.Field(
+        dtype=bool,
+        default=False,
+        doc="Set up and run diffim using settings that ensure the results"
+        "are compatible with the old version in pipe_tasks.",
+        deprecated="This option is only for backwards compatibility purposes"
+        " and will be removed after v24.",
+    )
+
+    def setDefaults(self):
+        self.makeKernel.kernel.name = "AL"
+        self.makeKernel.kernel.active.fitForBackground = self.doSubtractBackground
+        self.makeKernel.kernel.active.spatialKernelOrder = 1
+        self.makeKernel.kernel.active.spatialBgOrder = 2
+        if self.forceCompatibility:
+            self.mode = "convolveTemplate"
+
+
+class AlardLuptonSubtractTask(lsst.pipe.base.PipelineTask):
+    """Base class for image subtraction Tasks using the Alard & Lupton (1998)
```

Comment:
This is no longer a "base class". Maybe "Difference [or Subtract? or Compute the difference between?] a science and template image using the..."